roxie/roxiemem/roxiemem.cpp

```diff
@@ -461,6 +537,69 @@ extern void releaseRoxieHeap()
     }
 }
 
+extern void notifyAllUnusedRoxieMem()
+{
+    char *blockStart = 0;
+    char *blockEnd = 0;
+
+    CriticalBlock b(heapBitCrit);
+
+    if (heapHugePagesPossible)
+    {
+        // release only large chunks
+        for (unsigned i = 0; i < heapBitmapSize; i++)
+        {
+            heap_t t = heapBitmap[i];
+            if (t==HEAP_ALLBITS)
+            {
+                if (blockStart == 0)
+                {
+                    blockStart = heapBase + (i * HEAP_BITS * HEAP_ALIGNMENT_SIZE);
+                    blockEnd = blockStart + (HEAP_BITS * HEAP_ALIGNMENT_SIZE);
+                }
+                else
+                    blockEnd += (HEAP_BITS * HEAP_ALIGNMENT_SIZE);
+            }
+            else if (blockStart > 0)
+            {
+                notifyMemoryUnused(blockStart, memsize_t(blockEnd - blockStart));
+                blockStart = 0;
```

Comment:
style: assign nullptr.  Same comments about null initialization and comparison apply to the else branch.